src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -603,7 +603,10 @@ private void InitializeTrainingGraph(IDataView input)
                     (string)labelType.ToString());
             }
 
-            _classCount = labelCount == 1 ? 2 : (int)labelCount;
+            if (labelCount == 1)
+                throw new InvalidOperationException("Dataset contains only 1 class. ImageClassificationTrainer requires more than 1");
```

Comment:
>ImageClassificationTrainer [](start = 84, length = 26)

Could you make the message more generic. Its not only ImageClassification that requires more than 1 class. Pls see what other trainers throw and align the message with those

Reply:
So, for example, when fitting a `SdcaMaximumEntropy` trainer with a dataset with 1 label then `LinearMulticlassModelParametersBase` will throw this exception:
`System.ArgumentOutOfRangeException: 'Must be at least 2. Parameter name: numClasses'`

Because of this line of code:
https://github.com/dotnet/machinelearning/blob/c4e4263188dccf16903b8f3fea7e65213a69c6e3/src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs#L505

So I could change the exception in here to have the same output. But I didn't do that because, in general, I've seen users mentioning that our Exception messages aren't very readable. If I were to use the same exception message in here, it would be something like this:
`System.ArgumentOutOfRangeException: 'Must be at least 2. Parameter name: labelCount`

Which, to be honest, I think would be very unreadable for a user (specially one who is simply using the nugets, and wouldn't want to delve into the code to figure out what "labelCount" is and where is it used).

That's why I preferred to use the specific exception message I wrote, and also because it's an exception thrown in the ImageClassificationTrainer because of a requirement of the trainer itself, which throws before training.

Please let me know if you still think the exception message should be changed. If anything I guess something like
"Dataset contains only 1 class. More than 1 class is required. Parameter name: labelCount"

Reply:
@antoniovs1029 Good idea to make the exception message more readable. I think @ganik's point was that all classifiers need more than one class. But in this particular case, labelCount as a parameter would not mean anything to the user, since it is not a parameter that they supplied.

A good error message would also tell me what needs to be done to fix the problem. (Perhaps @natke can weigh in here about best practices) How about something like:

"Dataset contains only one class. Please use a dataset with more than one labeled class"

Reply:
I agree that consistency across all of the multiclass trainers would be good. But not at the expense of clarity if the existing messages are poorly written.

I had a look through the other exceptions and this is what I found:
- Linear trainers such as  SdcaMaximumEntropy: Unhandled Exception: System.ArgumentOutOfRangeException: Must be at least 2.
- LightGbm: "Number of classes should be specified and greater than 1 for multiclass training"
- Meta trainers such as OneVersusAll and Pairwise coupling don't even check for the condition, and let you build a model that always predicts the same class

To @harishsk's point about `labelCount` not being a parameter that the user supplied, can we reference the label column name from the schema, as that is what the user has supplied?

Something like: "Only one class found in XXX column. To build a multiclass classification model, the number of classes needs to be 2 or greater."

Reply:
Yes, we can reference the label column name to output that message. I will update this PR asap with those changes.

Reply:
So I've updated the PR with the exception message that @natke suggested.